mmv1/third_party/terraform/transport/config.go.tmpl

```diff
@@ -176,6 +178,56 @@ func (f *Formatter) Format(entry *logrus.Entry) ([]byte, error) {
 	return []byte(output), nil
 }
 
+type ExternalCredentialsHcpTerraform struct {
+	Audience            string
+	ServiceAccountEmail string
+	IdentityToken    string
+}
+
+var _ externalaccount.SubjectTokenSupplier = ExternalCredentialsHcpTerraform{}
+
+// SubjectToken returns the identity token passed to the provider as an argument from the config.
+// We do not interact with an external system to get a token.
+func (e ExternalCredentialsHcpTerraform) SubjectToken(ctx context.Context, options externalaccount.SupplierOptions) (string, error) {
+	if e.IdentityToken != "" {
+		return e.IdentityToken, nil
+	}
+	return "", errors.New("identity token unavailable in Config when configuring the provider")
```

Comment:
This is an example of an error we 'shouldn't' reach - this function should only be invoked if ExternalCredentialsHcpTerraform is set on the Config struct, i.e. the block is set in the provider block.

All the fields inside external_credentials_hcp_terraform are Required, so if external_credentials_hcp_terraform is present then the identity token should be present.